packages/dotcom-build-image-uploader/src/index.ts

```diff
@@ -0,0 +1,41 @@
+const path = require('path')
+const glob = require('glob')
+const MultiEntryPlugin = require('webpack/lib/MultiEntryPlugin')
+
+export class PageKitImageUploaderPlugin {
+  basePath: string
+
+  constructor(directory = './client') {
+    this.basePath = directory
+  }
+
+  apply(compiler) {
+    const imageFiles = glob.sync('**/*.{png,jpg,jpeg,gif,webp,ico,svg}', {
+      absolute: true,
+      cwd: path.resolve(this.basePath)
+    })
+
+    new MultiEntryPlugin(compiler.context, imageFiles, '__images__').apply(compiler)
+
+    const isDevMode = compiler.options.mode === 'development'
+    const outputFileName = isDevMode ? '[name].[ext]' : '[name].[contenthash:12].[ext]'
+
+    compiler.options.module.rules.push({
+      test: /\.(png|jpe?g|gif|webp|ico|svg)$/,
+      use: [
+        {
+          loader: require.resolve('file-loader'),
+          options: {
+            name: (resourcePath) => {
+              const dirname = path.dirname(resourcePath)
```

Comment:
I think it's worth adding some comments about why this is necessary (to maintain the relative path) and maybe an explicit test too.

Reply:
!["she doesn't even go here" meme](https://user-images.githubusercontent.com/631757/90656460-43b2e700-e23a-11ea-84a4-20797d7f5cbc.png)

Reply:
@i-like-robots regarding the explicit test for the relative path, this is already covered by the existing test where we check the assets that have been moved. Did you have something else in mind?